provider/machine_pool_resource.go

```diff
@@ -455,34 +456,11 @@ func (r *MachinePoolResource) Update(ctx context.Context, request tfsdk.UpdateRe
 		return
 	}
 
-	// AZ settings can't be changed. We're just calling validate to determine whether it's a multi-AZ pool or not.
-	isMultiAZPool, err := r.validateAZConfig(plan)
-	if err != nil {
-		response.Diagnostics.AddError(
-			"Can't build machine pool",
-			fmt.Sprintf(
-				"Can't build machine pool for cluster '%s': %v",
-				state.Cluster.Value, err,
-			),
-		)
-		return
-	}
-
 	computeNodesEnabled := false
 	autoscalingEnabled := false
 
 	if !plan.Replicas.Unknown && !plan.Replicas.Null {
 		computeNodesEnabled = true
-		if isMultiAZPool && plan.Replicas.Value%3 != 0 {
```

Comment:
Why did you remove this validation?
In case of multiple AZ, replicas must be replica of 3
This is the relevant validation in the cli:
https://github.com/openshift/rosa/blob/4de5ba8ee9b6b9eff715d3f8c5fa9ec362047584/cmd/edit/machinepool/machinepool.go#L65

Reply:
I'm the one that added the check originally: https://github.com/terraform-redhat/terraform-provider-rhcs/commit/bc3214e95efa633429ac62afceb5a4a9f80cd00a
OCM validates the replica count, so it isn't actually necessary... I just added it because (1) it was in the cli and (2) I had the info for other reasons.

I removed it because it relies on knowing `isMultiAZPool` which comes from `validateAZConfig()`, and the logic in the validate function wasn't able to cope w/ the state from import. It comes down to the validation using the user config values. However, import (Read) doesn't set them in the state, so TF believes the fields are null. Since they are locked from updating, the user can't set them (nor should they need to, honestly, since the values and validation only matter on Create, not Update).

I add a new function to determine the multi-ness of the pool and keep this validation if you'd like.

Reply:
But the validateAZConfig use the plan and not the state, so maybe it will work with the state?